lib/Producer.js

```diff
@@ -29,11 +29,16 @@ class Producer {
     if (this.lookupdCluster) {
       return this.lookupdCluster
         .nodes()
-        .map(node => {
-          return this.connectNsqd(node.broadcast_address, node.tcp_port, this.opts);
-        })
-        .then(() => {
-          return this.conns;
```

Comment:
No, I believe implementing it like this would address the root cause which is consistent to the error message.

> your implementation is not wrong, but will it be simpler to check the conns here?
> `throw if (conn.length == 0) `